Rewrite the Program Files article (#4874)
ppy/osu-wiki: wiki/osu!_Program_Files/en.md

### Data

osu! data files. Contains some of osu!'s cache, like beatmap background cache and avatar caches. They should not be deleted, because they may be in use by osu!.

### Files

*Caution: Be careful with these files, you might break osu! if you are not careful.*

## Database files (.db)

The database files are daabases that osu! requires to function propertly. The files contain vital information that osu! requires, such as saved scores, and the list of beatmaps saved on the player's device.

- `collections.db`: Stores the player's "Collections" in-game. The player can post them to the forums for others to see. Actual beatmaps are not provided with the collection file.
- `osu!.db`: Stores osu!'s database of beatmaps.
- `presence.db`: Stores a cache of osu!players logged in the Chat Console.
- `scores.db`: Stores the local leaderboards.

## .cfg (Configuration files)

Configuration files or config files configure the initial settings for osu! to work. The files can be opened with a text editor.

- `osu!.cfg`: Stores security information about the osu! application files and current release stream. This should never be modified manually.

> - `osu!.<username>.cfg`: Stores [Options](/wiki/Options) data and other game settings. See [User Configuration File](/wiki/osu!_Program_Files/User_Configuration_File).

Reviewer: ```suggestion
- `osu!.<operating system username>.cfg`: Stores [Options](/wiki/Options) data and other game settings. See [User Configuration File](/wiki/osu!_Program_Files/User_Configuration_File).
```

Author: Merged in 269856f

> - `osu!.<username>.cfg`: Stores [Options](/wiki/Options) data and other game settings. See [User Configuration File](/wiki/osu!_Program_Files/User_Configuration_File).
> 
> ### .exe (Application)
> 
> The main component. Click on it to start-up (only applies to Windows.) The .exe files are safe to open assuming the player used the osu!installer to install osu!.

Reviewer 2: ```suggestion
- `osu!.<username>.cfg`: Stores [Options](/wiki/Options) data and other game settings. See [User Configuration File](/wiki/osu!_Program_Files/User_Configuration_File).

### .exe (Application)

The main component. Click on it to start-up (only applies to Windows.) The .exe files are safe to open assuming the player used the osu!installer to install osu!. test
```
test comment, please ignore

# osu! program files

![The file structure of osu!'s installation folder, on Windows and macOS](img/file_structure.jpg "The file structure of osu!'s installation folder, on Windows and macOS")

The **osu! program files** are a set of files that run osu! and keep track of different activities as user's play.

## Installation paths

osu! is installed in the following locations, by default:

| Windows | macOS |
| :-- | :-- |
| `C:\Users\<Username>\AppData\Local\osu!` | `/Applications/osu!.app/Contents/drive_c/osu!` |

## Folders

### Chat

The Chat folder only appears if the player has "Automatically log private messages" enabled in Options, or the player runs the "/savelog" command in the [Chat Console](/wiki/Chat_Console).

> The name structure of the files is `{Tab_name}-{YYYYMMDD}-{HHMMSS}`, and can be opened in any text editor. An example of this is shown below:

Reviewer 3: structure? I think format is better

Author: What about this?

```suggestion
The naming scheme of the files is `{Tab_name}-{YYYYMMDD}-{HHMMSS}`, and can be opened in any text editor. An example of this is shown below:
```

*For further reference, please refer to the [Skinning](/wiki/Skinning) page.* 

### Songs

**The Songs folder holds the player's osu! beatmaps**. Usually contains `.osu` (difficulties), `.mp3`/`.ogg` (music files), `.jpg`/`.png`/`.gif` (background images), `.osb` (storyboard files) and `.mp4`/`.flv` (video files). May also contain `.wav`/`.ogg` (hitsound files) and folders (storyboard spites and/or skin folders).

The file name structure is `{Beatmap number} {Artist} - {Song Title}`.
**Example:** [57950 SOUND HOLIC - Drive My Life](https://osu.ppy.sh/beatmapsets/57950)

Please note that some very old beatmaps (for example, [Kenji Ninuma - DISCO PRINCE](https://osu.ppy.sh/beatmapsets/1) or [Dudelstudios - Angry Video Game Nerd Theme [MATURE CONTENT]](https://osu.ppy.sh/beatmapsets/66)), as well as unsubmitted beatmaps, do not follow the format.

## Hidden folders

These folders are hidden because any modifications to them could prevent osu! from starting correctly, or at all.

### Data

osu! data files. Contains some of osu!'s cache, like beatmap background cache and avatar caches. They should not be deleted, because they may be in use by osu!.

## Files

> *Caution: Be careful with these files, you might break osu! if you are not careful.*

Reviewer 4: The only change I would make is probably adding this caution to the songs folder to. Any user changes (not done via the osu! client) will potentially cause database corruption, or at least a long reprocess operation.

Author: Added in 60c1bfb

## Database files (.db)

> The database files are daabases that osu! requires to function propertly. The files contain vital information that osu! requires, such as saved scores, and the list of beatmaps saved on the player's device.

Reviewer 4: ```suggestion
The database files are databases that osu! requires to function propertly. The files contain vital information that osu! requires, such as saved scores, and the list of beatmaps saved on the player's device.
```

> The database files are daabases that osu! requires to function propertly. The files contain vital information that osu! requires, such as saved scores, and the list of beatmaps saved on the player's device.

Reviewer 3: ```suggestion
The database files are databases that osu! requires to function propertly. The files contain vital information that osu! requires, such as saved scores, and the cached list of beatmaps saved on the player's device.
```

Author: Done in 60c1bfb

> - `collections.db`: Stores the player's "Collections" in-game. The player can post them to the forums for others to see. Actual beatmaps are not provided with the collection file.

Reviewer 4: Since when has "posting" this file been a thing? I'd probably remove this.

Author: Remove the mention of collections.db, or the fact that you can post them online?

- `osu!.db`: Stores osu!'s database of beatmaps.
- `presence.db`: Stores a cache of osu!players logged in the Chat Console.
- `scores.db`: Stores the local leaderboards.

## .cfg (Configuration files)

> Configuration files or config files configure the initial settings for osu! to work. The files can be opened with a text editor.

Reviewer 3: ```suggestion
Configuration files configure the initial settings for osu! to work. The files can be opened with a text editor.
```

Author: Fixed in 12234dc

- `osu!.cfg`: Stores security information about the osu! application files and current release stream. This should never be modified manually.
- `osu!.<operating system username>.cfg`: Stores [Options](/wiki/Options) data and other game settings. See [User Configuration File](/wiki/osu!_Program_Files/User_Configuration_File).

### .exe (Application)

> The main component. Click on it to start-up (only applies to Windows.) The .exe files are safe to open assuming the player used the osu!installer to install osu!.

Reviewer 3: ```suggestion
The main component. Click on it to start-up (only applies to Windows.) The .exe files are safe to open assuming the player used the osu!installer downloaded from the official website to install osu!.
```

Author: Should be done in e258f5a

osu!.exe (Start-up osu!)

#### .dll (application extension)

> These .dll files are osu!'s components and dependencies.

Reviewer 4: Not sure of the value of including this. It's already outdated.

Author: I think maybe the list should go, but the text that describes what they're for should stay.

Author: Done in 57ca705

> # osu! program files

Reviewer 2: ```suggestion
# osu! program files

*See also: [osu! File Formats](/wiki/osu!_File_Formats)*
```

Author: Added in 0aab9e0

![The file structure of osu!'s installation folder, on Windows and macOS](img/file_structure.jpg "The file structure of osu!'s installation folder, on Windows and macOS")

The **osu! program files** are a set of files that run osu! and keep track of different activities as user's play.

## Installation paths

> osu! is installed in the following locations, by default:

Reviewer 2: ```suggestion
By default, osu! is installed in the following locations:
```
still not too hot on the use of a table here

Author: Fixed in 18b1c34

| Windows | macOS |
| :-- | :-- |
| `C:\Users\<Username>\AppData\Local\osu!` | `/Applications/osu!.app/Contents/drive_c/osu!` |

## Folders

### Chat

The Chat folder only appears if the player has "Automatically log private messages" enabled in Options, or the player runs the "/savelog" command in the [Chat Console](/wiki/Chat_Console).

The name structure of the files is `{Tab_name}-{YYYYMMDD}-{HHMMSS}`, and can be opened in any text editor. An example of this is shown below:

> **Example**: #multiplayer-20121115-040845 (/savelog at #multiplayer tab in 15th November 2012 at 04hrs, 08mins, and 45secs).

Reviewer 2: if you have an "an example is shown below" up top, you dont need to have this here. this line should also be surround by grave marks as well

Author: Fixed in 49d39e3

### Downloads

The Downloads folder holds the beatmaps being downloaded by osu!direct (requires [osu!supporter](/wiki/osu!supporter)). They get transferred to the Songs folder upon completion.

### Exports

The Exports folder appears if the player uses the [Skin Selector's "Export as .osk"](/wiki/Options) or [Beatmap Editor's "Export Package"](/wiki/Beatmap_Editor/Menu). It will hold the beatmaps and skins the player has exported from osu!.

> Players that want to know how to do this can see [osu! File Formats](/wiki/osu!_File_Formats).

Reviewer 2: this probably isnt neccessary

Author: Addresed in 0aab9e0

The Replays folder holds the player's replay files. A replay file does not work when the beatmaps linked to it is missing. The replay also contains the results data, and reanimates the player's cursor movement while replaying. To create a replay, press F2 at the results screen, or click on the 'Save replay to Replays folder' (in Solo only). 

*For players who interested in uploading their replay to YouTube, see: [Osr2mp4 public release. Automatically convert replay file to video.](https://osu.ppy.sh/community/forums/topics/1104243)

The file name structure is `{Local player name} - {Artist} - {Title} {[Difficulty]}{(YYYY-MM-DD)} {Game Mode}`. An example of this is shown below:

**Example:** dummytest1 - Loituma - Ievan Polkka \[SPINNER-MADNESS\]  (2013-08-12) OsuMania

### Screenshots

The Screenshots folder holds screenshots the player has created in osu!. By default, the saved screenshot's file extension is `.jpg`, however this can be changed to `.png` in the Options menu.

*Note: To create a screenshot, press the screenshot key (F12 by default).*

The file name structure is `screenshot###`, where "###" is the screenshot number count.

### Skins

> The Skins folder holds **user-created skins, which can be used to customise the in-game interface.** Players can download skins from the [Skinning subforum](https://osu.ppy.sh/community/forums/15). Players can install skins by double-clicking on the skin from a file manager. "osu! by peppy" is the only skin without its folder and cannot be deleted.

Reviewer 2: any reason why this is bolded here?

Author: Whoops, forgot about that!

> *For further reference, please refer to the [Skinning](/wiki/Skinning) page.* 

Reviewer 2: should be a hatnote

Author: According to the ASC, it's a "For see" hatnote 
https://osu.ppy.sh/wiki/en/Article_styling_criteria/Formatting#see-also

> # osu! program files

Author: This has been fixed since e384c73

> ![The file structure of osu!'s installation folder](img/osu!-program-files.png "The file structure of osu!'s installation folder")

Author: > 
> 
> would you by any chance have access to someone who can take images of these folders for macOS? this isn't immediately neccessary but would be nice to have

I'll see what I can do about this

> ![The file structure of osu!'s installation folder](img/osu!-program-files.png "The file structure of osu!'s installation folder")

Author: I'll have the screenshot ready tomorrow (local time), but it's already referred to in the wiki, all I have to do is crop and upload.

> ![The file structure of osu!'s installation folder](img/osu!-program-files.png "The file structure of osu!'s installation folder")

Author: Screenshot has been added

### Installation Paths ###
<!--TODO: Find out where macOS osu! is stored-->
osu! is installed by default in the following locations, by default:
| Windows 	| C:\Users\<username>\AppData\Local\osu! 	  |
|---------	|----------------------------------------	  |

> | macOS     | /Applications/osu!.app/Contents/drive_c/osu!|

Author: I pushed the wrong branch, should be fixed now.

## Folders

> ### Chat

Author: Can you give an example on what you mean, because I'm not sure I'm understanding.

> ### Chat

Reviewer 2: Nvm, I just checked, my comment isnt relevant

> This folder only appears if the player has "Automatically log private messages" enabled in Options, or the player runs the "/savelog" command in the [Chat Console](/wiki/Chat_Console).

Author: Added grave marks in f7c5037

> The file name structure is `{Tab_name}-{YYYYMMDD}-{HHMMSS}`, and can be opened in any text editor (e.g. Notepad)

Author: Whoops, this one's not done yet

> The file name structure is `{Tab_name}-{YYYYMMDD}-{HHMMSS}`, and can be opened in any text editor (e.g. Notepad)

Author: Done in 9772127

**Example**: #multiplayer-20121115-040845 (/savelog at #multiplayer tab in 15th November 2012 at 04hrs, 08mins, and 45secs).

### Downloads

This folder holds the beatmaps being downloaded by osu!direct (requires [osu!supporter](/wiki/osu!supporter)). They get transferred to the Songs folder upon completion.

### Exports

This folder appears if the player uses the [Skin Selector's "Export as .osk"](/wiki/Options) or [Beatmap Editor's "Export Package"](/wiki/Beatmap_Editor/Menu). It will hold the beatmaps and skins the player has exported from osu!.

Players that want to know how to do this can see [osu! File Formats](/wiki/osu!_File_Formats).

### Localisations

> This folder appears when the player has switched their language in the options. It will hold the translated text files which can be used each to replace the English text based on the user's selected localisation.

Author: The text file gets generated when you switch languages.

### Replays

> This folder holds the player's replay files. A replay file does not work when the beatmaps linked to it is missing. The replay also contains the results data, and reanimates the player's cursor movement while replaying. To create a replay, press F2 at the results screen, or click on the 'Save replay to Replays folder' (in Solo only). [For players who are interested to upload their replay to the YouTube, see this thread](https://osu.ppy.sh/community/forums/topics/1104243).

Author: Added a note in 269856f

The format is `{Local player name} - {Artist} - {Title} {[Difficulty]}{(YYYY-MM-DD)} {Game Mode}`

> **Example:** dummytest1 - Loituma - Ievan Polkka \[SPINNER-MADNESS\]  (2013-08-12) OsuMania

Author: Fixed in 9772127

### Screenshots

> **This folder holds screenshots the player has created in osu!**. The saved screenshot's file extension (.jpg/.png) is based on what the player set on the Options menu.

Author: Added @MilkyIQ's suggested change in 9772127

> **This folder holds screenshots the player has created in osu!**. The saved screenshot's file extension (.jpg/.png) is based on what the player set on the Options menu.

Author: Should I unbold it?

> **This folder holds screenshots the player has created in osu!**. The saved screenshot's file extension (.jpg/.png) is based on what the player set on the Options menu.

Author: Unbolded in fbca03b

To create a screenshot, press the screenshot key (F12 by default).

The file name structure is `screenshot###`, where "###" is the screenshot number count.

### Skins

<<<<<<< HEAD
This folder holds **user-created skins, which can be used to customise the in-game interface.** Players can download skins from the [Skinning subforum](https://osu.ppy.sh/community/forums/15). Players can install skins by double-clicking on the skin from a file manager. 
=======
This folder holds **user-created skins, which can be used to customise the in-game interface.** You can download skins from the [Skinning subforum](https://osu.ppy.sh/community/forums/15). You can install .osk skins by double-clicking them. If the downloaded skin is in folder form, you will have to place the folder here yourself. If it is in a ZIP or RAR format, you must extract it first. You can change your skins at [Options menu under Skins tab (Skin Selector)](/wiki/Options). Please bear in mind that the selected skin is only visible to you.
>>>>>>> e78270d1d920905df8e145d2cccd9e5ab8864daa

> For further reference, please refer to the [Skinning](/wiki/Skinning) page. Also, "osu! by peppy" is the only skin without its folder and cannot be deleted.

Author: Is now a hatnote in 4254534

### Songs

> **This folder holds the player's osu! beatmaps**. Usually contains .osu (difficulties), .mp3/.ogg (music files), .jpg/.png/.gif (background images), .osb (storyboard files) and .mp4/.flv (video files). May also contains .wav/.ogg (hitsound files) and folders (storyboard spites and/or skin).

Reviewer 2: any file extension, full file name, or filepath should be surrounded in grave marks

Author: Added grave marks in f7c5037

The file name structure is `{Beatmap number} {Artist} - {Song Title}`.
**Example:** [57950 SOUND HOLIC - Drive My Life](https://osu.ppy.sh/beatmapsets/57950)

Please note that some very old beatmaps (for example, [Kenji Ninuma - DISCO PRINCE](https://osu.ppy.sh/beatmapsets/1) or [Dudelstudios - Angry Video Game Nerd Theme [MATURE CONTENT]](https://osu.ppy.sh/beatmapsets/66)), as well as unsubmitted beatmaps, do not follow the format.

> ## Hidden folders

Reviewer 2: "Hidden folders" needs a short description

also the rest of these sections need more information in order to justify splitting them up into separate sections. Otherwise, the information should be condensed into paragraph or two

Author: Added a short description to the Hidden folders in cd34b18
